---
title: "Load packages and data"
format: html
---

# LOAD PACKAGES
```{r}
library(pacman)
pacman::p_load(pacman, EBSeq, ggpubr, tidyverse, RColorBrewer, here, vroom, tictoc)
```
# -------------------------------

# IMPORT DATA

## Annotations
```{r}
tic()
annot = vroom(file = here::here("data", "raw", "annotations.tsv")) # 29.5 secs
toc()
colnames(annot)
annot = as.data.frame(annot)
n_distinct(annot$`#contig_id`) # 10,303,792 distinctive contigs
```

## NA degradation genes annotations (generated in MG)
```{r}
tic()
annot_NA = vroom(file = here::here("data", "clean", "degradation_gene_annotations.csv")) # 0.38 secs - 18,272 obs in 34 vars
toc()
```


## MT data
```{r}
MT = vroom(file = here::here("data", "raw", "01_merged_contig_abundance_cpm.tsv")) # 18.5 secs
head(MT)
colnames(MT)
MT = as.data.frame(MT)
View(MT[1:5,1:10])
rownames(MT) = MT$...1 # get contig code as rownames
```

# -------------------------------
# NA DEGRADATION CONTIGS

##Inner join NA degradation
```{r}
names(MT)[names(MT) == "...1"] = "contig_id" # Rename the contig_id variable from ...1 column in MT to match value in annot_NA and do inner_join with this variable.
#MT$contig_id = MT$...1 
#MT$...1 = NULL # Erase ...1
MT_NA_full = left_join(x = annot_NA, y = MT) # Find NA degradation gene transcript contigs
MT_NA = MT_NA_full[, grepl("NEB", names(MT_NA_full))] # keep only samples
MT_NA = cbind(MT_NA_full$contig_id, MT_NA) # attach contig_id
```

## Find contigs with rowSum > 0 (contig present in at least one sample)
```{r}
rowsum = as.data.frame(rowSums(MT_NA[sapply(MT_NA, is.numeric)])) # Calculate rowsums 
nonzeros = apply(rowsum, 2, function(c)sum(c!=0)) # identify number of non-zero values. 17,560 out of 18,272 contigs is found in MT compared to MG.
ratio.present.NA.transcripts = nonzeros / nrow(rowsum) * 100 # 96.1 NA degradation genes identified in MG were found in MT data set. 
```

## Keep only NA contigs with presence in at least one sample
```{r}
MT_NA_pstv = MT_NA[rowSums(MT_NA[sapply(MT_NA, is.numeric)]) > 0,] # filter
rowsum = as.data.frame(rowSums(MT_NA_pstv[sapply(MT_NA_pstv, is.numeric)])) # Check that there are no zeros
```

## Filter out in Annot_NA the contigs not selected in MT_NA_pstv
```{r}
annot_NA_pstv = annot_NA |> filter(contig_id %in% MT_NA_pstv$`MT_NA_full$contig_id`) # 17,560 obs in 34 vars
```











# -------------------------------

# DATE
## D0
```{r}
MT_D0 = MT[, grepl("D-0", names(MT))]# 10,360,835 contigs in 34
MT_D0 = MT_D0[rowSums(MT_D0 > 0),] # 7,096,181 contigs in 34 variables
vroom::vroom_write(MT_D0, file = here::here("data", "clean", "D-0_merged_contig_abundance_cpm.tsv"))

annot_D0 = 
```

## D8
```{r}
MT_D8 = MT[, grepl("D8", names(MT))]# 10,360,835 contigs in 41
MT_D8 = MT_D8[rowSums(MT_D8 > 0),] # 7,096,181 contigs in 34 variables
vroom::vroom_write(MT_D8, file = here::here("data", "clean", "D8_merged_contig_abundance_cpm.tsv"))
```
